polaris-react/src/components/TextField/TextField.scss

```diff
@@ -249,6 +251,23 @@ $spinner-icon-size: 12px;
   }
 }
 
+.borderlessSlim {
+  .Input,
+  .Backdrop {
+    border: none;
+    min-height: 28px;
+    padding-block: var(--p-space-050);
+  }
+}
+
+.slim {
+  .Input,
+  .Backdrop {
+    min-height: 28px;
+    padding-block: var(--p-space-050);
+  }
+}
+
```

Comment:
Note `Filters` search field height:
|-|medium and up screens|small screens|
|-|-|-|
|[Production](https://storybook.polaris.shopify.com/?path=/story/all-components-filters--with-a-resource-list)|`28px`|`28px`|
|[Figma](https://www.figma.com/file/jLLkmo9r28hiqPvf4s90E4/Polaris-Gen-3-Components?type=design&node-id=67128%3A21676&mode=design&t=0yMmrHC8rlYKmHQq-1)|`28px`|`36px`|
|This PR|`28px`|`28px`|

let me know if I should add a different height for small screens here. `36px` seemed drastically bigger than what is in production today but we can always update